File: src/main/kotlin/kotlin99/P16.kt
package kotlin99

/**
 * Created by rick on 2017/2/20.
 */
fun drop(count: Int, list: List<Char>): List<Char> {
    if (list.isEmpty())
        return emptyList()
    else if (count > list.size){
        return list
    } else {
        return list.filterIndexed { index, data ->
            (index + 1) % count != 0
        }
    }
}

File: src/main/scala/scala99/P16.scala
package scala99

/**
  * Created by rick on 3/31/2017.
  */
object P16 {
    def drop[T](i: Int, list: List[T]): List[T] = {
        list.zipWithIndex filter { x => ((x._2 + 1) % i) != 0 } map { x => x._1 }
    }
}
